File: py/IEC61968/InfIEC61968/InfERPSupport/ErpLedgerEntry.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from IEC61968.Common.Status import Status
from IEC61968.Common.UserAttribute import UserAttribute
from IEC61968.InfIEC61968.InfERPSupport.ErpAccountKind import ErpAccountKind
from IEC61968.InfIEC61968.InfERPSupport.ErpIdentifiedObject import ErpIdentifiedObject
from IEC61968.InfIEC61968.InfERPSupport.ErpJournalEntry import ErpJournalEntry
from IEC61968.InfIEC61968.InfERPSupport.ErpLedBudLineItem import ErpLedBudLineItem
from IEC61970.Base.Domain.DateTime import DateTime
from IEC61970.Base.Domain.Money import Money


class ErpLedgerEntry(ErpIdentifiedObject):
    
    def __init__(self):
        super().__init__()
        self.account_id = ""
        self.account_kind = ErpAccountKind.NORMAL
        self.amount = Money()
        self.posted_date_time = DateTime()
        self.status = Status()
        self.transaction_date_time = DateTime()
        self.erp_journal_entry = ErpJournalEntry()
        self.erp_ledger_entry = ErpLedBudLineItem()
        self.user_attributes = UserAttribute()

File: java/IEC61968/InfIEC61968/InfERPSupport/ErpLedgerEntry.java
import IEC61970.Base.Domain.String;
import IEC61970.Base.Domain.Money;
import IEC61970.Base.Domain.DateTime;
import IEC61968.Common.Status;
import IEC61968.Common.UserAttribute;

/**
 * Details of an individual entry in a ledger, which was posted from a journal on
 * the posted date.
 * @created 25-Dec-2023 8:45:21 PM
 */
public class ErpLedgerEntry extends ErpIdentifiedObject {

	/**
	 * Account identifier for this entry.
	 */
	public String accountID;
	/**
	 * Kind of account for this entry.
	 */
	public ErpAccountKind accountKind;
	/**
	 * The amount of the debit or credit for this account.
	 */
	public Money amount;
	/**
	 * Date and time this entry was posted to the ledger.
	 */
	public DateTime postedDateTime;
	public Status status;
	/**
	 * Date and time journal entry was recorded.
	 */
	public DateTime transactionDateTime;
	public ErpJournalEntry ErpJounalEntry;
	public ErpLedBudLineItem ErpLedgerEntry;
	public UserAttribute UserAttributes;

	public ErpLedgerEntry(){

	}
}//end ErpLedgerEntry